src/Microsoft.AspNetCore.Sockets/Transports/LongPollingTransport.cs

```diff
@@ -37,18 +37,19 @@ public LongPollingTransport(IReadableChannel<Message> connection, ILoggerFactory
                 // TODO: We need the ability to yield the connection without completing the channel.
                 // This is to force ReadAsync to yield without data to end to poll but not the entire connection.
                 // This is for cases when the client reconnects see issue #27
-                using (var message = await _connection.ReadAsync(context.RequestAborted))
+                await _application.WaitToReadAsync(context.RequestAborted);
```

Comment:
Do we care about the return value? If there are no more messages and the channel is closed during this await the request will fall through the below and end up... 404ing?